src/webviews/settings/index.ts

```diff
@@ -381,8 +380,8 @@ export class SettingsUI {
                 delete data.password;
                 delete data.buttons;
 
-                connections[connectionIdx] = Object.assign(connection, data);
-                await GlobalConfiguration.set(`connections`, connections);
+                stored = Object.assign(stored, data);
+                await ConnectionManager.updateByIndex(index, stored);
```

Comment:
We should refresh the browser here to update the connection node name/description.
```suggestion
                await ConnectionManager.updateByIndex(index, stored);
                vscode.commands.executeCommand(`code-for-ibmi.refreshConnections`);
```